annlite/executor.py

```diff
@@ -58,8 +60,11 @@ def __init__(
             raise ValueError('Please specify the dimension of the vectors to index!')
 
         self.metric = metric
-        self.limit = limit
+        self.match_args = match_args
         self.include_metadata = include_metadata
+        self.limit = limit
+        if 'limit' in self.match_args:
+            self.limit = self.match_args.get('limit')
```

Comment:
Let's just keep `limit` in the arguments to overwrite `match_args`. 
```suggestion
        if limit:
            self.match_args.update('limit', limit)
```

Reply:
And the default value of limit also kept?

Reply:
yes, I think so.